packages/financial-templates-lib/src/clients/FinancialContractClient.js

```diff
@@ -159,9 +165,35 @@ class FinancialContractClient {
     ]);
 
     if (this.contractType === "Perpetual") {
-      this.latestCumulativeFundingRateMultiplier = this.toBN(
-        (await this.financialContract.methods.fundingRate().call()).cumulativeMultiplier.rawValue
-      );
+      if (!this.multicallContractAddress) {
+        // If no multicontract address set, then read the contract's on-chain CFRM. Note
+        // that this will not take into account any pending fundingrates that will be
+        // published on the next contract interaction.
+        this.latestCumulativeFundingRateMultiplier = this.toBN(
+          (await this.financialContract.methods.fundingRate().call()).cumulativeMultiplier.rawValue
+        );
+      } else {
+        // Simulate calling `applyFundingRate()` on the perpetual before reading
+        // `fundingRate()`, in order to account for any unpublished, pending funding rates.
+        const applyFundingRateCall = {
+          target: this.financialContract.options.address,
```

Comment:
is this not accessible directly from `.address` or `._address`? going via options is a bit different to what we normally do, no?

Reply:
Changed to `this.financialContractAddress`

Reply:
@chrismaree this is how you access the address for web3 contracts I believe. Truffle contracts offer the .address I think.

Reply:
ye, this is a web3 contract as we are within the bots, not in a truffle test. both `.options.address` and `._address` are bound to the same thing. Looks like we actually use both methods within the repo if you search for the strings :)